lib/std/os.zig

```diff
@@ -441,6 +441,7 @@ fn getRandomBytesDevURandom(buf: []u8) !void {
 /// Causes abnormal process termination.
 /// If linking against libc, this calls the abort() libc function. Otherwise
 /// it raises SIGABRT followed by SIGKILL and finally lo
+/// assume: Current signal handler for SIGABRT does **not call abort**.
```

Comment:
Don't assume this.

Reply:
This is what musl assumes. I checked llvm-libc and found a hint to why glibc does the recursive locking+unlocking dance.

There are 2 ways to "fix" this:
1. do the Linux thing and prepare user to always assume the user-installed handler is run to force the user to handle the possible recursion
2. do the glibc recursive locking+unlocking dance, with a mutex per thread which has the same drawbacks of potential failure in abort() and thus broken process teardown:
```txt
      /* This stage is special: we must allow repeated calls of
	 `abort' when a user defined handler for SIGABRT is installed.
	 This is risky since the `raise' implementation might also
	 fail but I don't see another possibility.  */
```

For a more elaborative discussion, see https://stackoverflow.com/q/187761/9306292
Also, as far as I am aware, libstd does not support recursive mutexes (part from Mutex.zig):
```txt
/// It is undefined behavior if the mutex is already held by the caller's thread.
```